src/Confluent.SchemaRegistry.Serdes.Avro/AvroUtils.cs

```diff
@@ -55,19 +55,14 @@ public static async Task<object> Transform(RuleContext ctx, Avro.Schema schema,
                     return await Transform(ctx, us[unionIndex], message, fieldTransform).ConfigureAwait(false);
                 case Avro.Schema.Type.Array:
                     ArraySchema a = (ArraySchema)schema;
-                    var arrayTasks = ((IList<object>)message)
-                        .Select(it => Transform(ctx, a.ItemSchema, it, fieldTransform))
-                        .ToList();
-                    object[] items = await Task.WhenAll(arrayTasks).ConfigureAwait(false);
-                    return items.ToList();
+                    var arrayTransformer = (int index, object elem) =>
+                        Transform(ctx, a.ItemSchema, elem, fieldTransform);
+                    return await Utils.TransformEnumerableAsync(message, arrayTransformer).ConfigureAwait(false);
                 case Avro.Schema.Type.Map:
                     MapSchema ms = (MapSchema)schema;
-                    var dictTasks = ((IDictionary<object, object>)message)
-                        .Select(it => Transform(ctx, ms.ValueSchema, it.Value, fieldTransform)
-                            .ContinueWith(t => new KeyValuePair<object, object>(it.Key, it.Value)))
-                        .ToList();
-                    KeyValuePair<object, object>[] entries = await Task.WhenAll(dictTasks).ConfigureAwait(false);
-                    return entries.ToDictionary(it => it.Key, it => it.Value); 
+                    var mapTransformer = (object key, object value) =>
+                        Transform(ctx, ms.ValueSchema, value, fieldTransform);
+                    return await Utils.TransformDictionaryAsync(message, mapTransformer).ConfigureAwait(false);
```

Comment:
You're changing the behavior slightly. Previously, all transformations were done in parallel (if there was any asynchrony in place). With this change, you will transform each element of the array sequentially.

Reply:
Yes, thanks, I'm keeping things simple for now.

Reply:
Ok, modified to use Task.WhenAll